include/host_queue.h

```diff
@@ -151,24 +153,26 @@ namespace detail {
 		 * @brief Waits until all currently submitted operations have completed.
 		 */
 		void wait() {
-			for(auto& ct : m_threads) {
-				ct.second.thread.stop(true /* isWait */);
+			const std::lock_guard lock(m_mutex); // called by main thread - never contended because the executor is shut down at this point
+			for(auto& [cgid, cp] : m_threads) {
+				cp.pool.stop(true /* isWait */);
```

Comment:
Why `cp` instead of `ct`?

Reply:
Im destructuring the `ct` that we iterated over before, and the right-hand side is **c**gid + **p**ool.